test/integration/acceptance/custom/hello_policy/hostnames_test.go

```diff
@@ -0,0 +1,461 @@
+func testHostnamesPolicy(t *testing.T, pub, prv *base.ClusterContext) {
+
+	// Normal init, plus getting hostname information from temporary link
+	init := []policyTestCase{
+		{
+			name: "init",
+			steps: []policyTestStep{
+				{
+					name:     "execute",
+					parallel: true,
+					cliScenarios: []cli.TestScenario{
+						skupperInitInteriorTestScenario(pub, "", true),
+						skupperInitEdgeTestScenario(prv, "", true),
+					},
+				}, {
+					prvPolicy: []skupperv1.SkupperClusterPolicySpec{allowedOutgoingLinksHostnamesPolicy("*", []string{"*"})},
+					name:      "create-token-link",
+					cliScenarios: []cli.TestScenario{
+						createTokenPolicyScenario(pub, "prefix", testPath, "hostnames", true),
+						// This link is temporary; we only need it to get the hostnames for later steps
+						createLinkTestScenario(prv, "", "hostnames", false),
+						linkStatusTestScenario(prv, "", "hostnames", true),
+					},
+				}, {
+					// We need to know the actual hosts we'll be connecting to, so we get them from the secret
+					name: "register-hostnames",
+					preHook: func(context map[string]string) error {
+						secret, err := prv.VanClient.KubeClient.CoreV1().Secrets(prv.Namespace).Get("hostnames", v1.GetOptions{})
+						if err != nil {
+							return err
+						}
+						url, err := url.Parse(secret.ObjectMeta.Annotations["skupper.io/url"])
+						if err != nil {
+							return err
+						}
+						host, _, err := net.SplitHostPort(url.Host)
+						if err != nil {
+							return err
+						}
+						log.Printf("registering claim host = %v", host)
+						context[originalClaim] = host
+
+						interRouterHost, ok := secret.ObjectMeta.Annotations["inter-router-host"]
+						if !ok {
+							return fmt.Errorf("inter-router-host not available from secret")
+						}
+						log.Printf("registering router host = %v", interRouterHost)
+						context[originalRouter] = interRouterHost
+
+						edgeHost, ok := secret.ObjectMeta.Annotations["edge-host"]
+						if !ok {
+							return fmt.Errorf("edge-host not available from secret")
+						}
+						log.Printf("registering edge host = %v", edgeHost)
+						context[originalEdge] = edgeHost
+
+						return nil
+					},
+				}, {
+					name:      "remove-tmp-policy-and-link",
+					prvPolicy: []skupperv1.SkupperClusterPolicySpec{allowedOutgoingLinksHostnamesPolicy("REMOVE", []string{})},
+					cliScenarios: []cli.TestScenario{
+						linkStatusTestScenario(prv, "", "hostnames", false),
+						linkDeleteTestScenario(prv, "", "hostnames"),
+					},
+				},
+			},
+		},
+	}
+
+	cleanup := []policyTestCase{
+		{
+			name: "cleanup",
+			steps: []policyTestStep{
+				{
+					name:     "delete",
+					parallel: true,
+					cliScenarios: []cli.TestScenario{
+						deleteSkupperTestScenario(pub, "pub"),
+						deleteSkupperTestScenario(prv, "prv"),
+					},
+				},
+			},
+		},
+	}
+
+	// a policyTestRunner table will be generated from these
+	tests := []hostnamesPolicyInstructions{
```

Comment:
I would say, we can keep it as is for now. And with regards to adding extra env vars, my preference
is to use build tags. That is how we are filtering tests today, so if we introduce an env var it will start
to get confusing, imo.